OpenSim/Actuators/Test/testMuscleFixedWidthPennationModel.cpp

```diff
@@ -795,13 +796,38 @@ int main(int argc, char* argv[])
         cout << "**************************************************" << endl;
         cout << "TEST: Exception Handling" << endl;
 
-        //constructor
-        SimTK_TEST_MUST_THROW(MuscleFixedWidthPennationModel fibKinEX = 
-            MuscleFixedWidthPennationModel(0, optPenAng, SimTK::Pi/2.0));
-        SimTK_TEST_MUST_THROW(MuscleFixedWidthPennationModel fibKinEX = 
-            MuscleFixedWidthPennationModel(optFibLen, -0.01, SimTK::Pi/2.0));
-        SimTK_TEST_MUST_THROW(MuscleFixedWidthPennationModel fibKinEX = 
-            MuscleFixedWidthPennationModel(optFibLen, SimTK::Pi/2, SimTK::Pi/2.0));
+        // Test property bounds.
+        {
+            MuscleFixedWidthPennationModel mfwpm;
+            mfwpm.set_optimal_fiber_length(0.0);
+            SimTK_TEST_MUST_THROW_EXC(mfwpm.finalizeFromProperties(),
+                SimTK::Exception::ErrorCheck);
+        }
+        {
+            MuscleFixedWidthPennationModel mfwpm;
+            mfwpm.set_pennation_angle_at_optimal(-SimTK::SignificantReal);
+            SimTK_TEST_MUST_THROW_EXC(mfwpm.finalizeFromProperties(),
+                SimTK::Exception::ValueOutOfRange);
+        }
+        {
+            MuscleFixedWidthPennationModel mfwpm;
+            mfwpm.set_pennation_angle_at_optimal(SimTK::Pi/2.0);
+            SimTK_TEST_MUST_THROW_EXC(mfwpm.finalizeFromProperties(),
+                SimTK::Exception::ValueOutOfRange);
+        }
+        {
+            MuscleFixedWidthPennationModel mfwpm;
+            mfwpm.set_maximum_pennation_angle(-SimTK::SignificantReal);
+            SimTK_TEST_MUST_THROW_EXC(mfwpm.finalizeFromProperties(),
+                SimTK::Exception::ValueOutOfRange);
+        }
+        {
+            MuscleFixedWidthPennationModel mfwpm;
+            mfwpm.set_maximum_pennation_angle(SimTK::Pi/2.0
+                                              + SimTK::SignificantReal);
```

Comment:
Is Pi/2 a legal value? Do you we allow zero (`cos(Pi/2)`) fiber force to be transmitted to the tendon?